### PR TITLE
This fixes #112 and beyond by now introducing proper SI unit types

### LINKED ISSUE #112: Default voltage values way out in PSAT models (and possible more)
Just noticed that OpenIPSL's insistence on non SI base of k and M values instead of the real voltage and power values caused that many default values are too large by a factor of 1000. Reason is that whoever added the values in the first place thought of them (rightly so in my opinion) as Volts and are computational treated as kV. 

Example from OpenIPSL.Electrical.Branches.PSAT.TwoWindingTransformer`:

```mo
parameter Real Vn=400000 "Voltage rating (kV)"
```

There are two ways to solve this.
 1. (the quick and dirty and not modelica really): change all the default parameters to be of kV and M
 2. (the cleaner and you **REALLY** should consider this): Get rid of any implicit k and M values and treat parameters in there base units. Add displayUnits if you like them to be displayed otherwise but as is shown already in the current state even the original developers (who argue all the time that they only think in kV and M because of the domain of the library) managed to make a mess out of this. 

*The only reason this did not lead to disaster (yet) was that voltages were always used in a scaling fashion. So the error of 1000 was self-correcting. But I would not rely on this.

### PR DESCRIPTION
**Pull request by dietmarw**
- Uses `displayUnit` instead of some arbitrary Reals. Supported by OMEdit and Dymola
-  Users can still enter and see the units in kV and MVA, Mvar, MW but all equations are now using the root SI units 
 - Solar and Wind were not touched yet though not a big impact since SI units were mainly used for scaling
 - Whenever sensible the convention of lower case variable/parameter names representing per-unit values and upper case representing SI units. Was done mainly for user facing parameters
 - This PR also provides a conversion script that should take care of all class name changes and modifier changes incl scaling
 - Since parameter scaling was random in the past there is a danger that existing user models get the wrong scale (in case the conversion script does not catch it). Normally easy to spot by simulation errors since variables suddenly differ by one or two order of magnitudes.
- As a last step all annotations where cleaned up.

### CODE REVIEW

**Comment by dietmarw** [comment]
Mind the check will fail because of the non-updated ApplicationExamples. I don't want to touch them now but do them one by one **after** this PR is merged.

**Comment by dietmarw** [comment]
I just remembered that what I could do is to uncomment the Travis checks on ApplicationExamples for now, so we get the OpenIPSL checked itself. Then when working and maybe moving the examples around we can reactivate the tests for the updated examples. Would that be an idea @marcelofcastro

**Comment by marcelofcastro** [comment]
@dietmarw yeah that's a good idea. Is it ok if I push some commits  to this branch later while reviewing this PR? Otherwise I can review it and post what may be modified (like I did last time).

**Comment by dietmarw** [comment]
Hi, pushing to this PR is perfectly fine with me. I'm now testing the conversion on some student models and see that it also needs to get tweaked a bit.

**Comment by dietmarw** [comment]
After some more testing the real remaining issue seems really to be getting the conversion rules right. One problem here is that there are issues that need to get resolved on the tool side (currently only Dymola) which will not happen before after the upcoming Modelica 2019 conference. 

But that should not keep us merging this branch as soon as the review of OpenIPSL is done. Otherwise we end up having more problems down the line. We can then work on adapting the ApplicationExamples one by one. After all the master is not the released version :-)

File: OpenIPSL/Electrical/Buses/BusExt.mo [hunk 1]
```diff
@@ -26,19 +26,20 @@ model BusExt
...
-  parameter Real V_0=1 "Voltage magnitude (pu)"
+  SI.PerUnit v(start=v_0) "Bus voltage magnitude (pu)";
+  SI.Conversions.NonSIunits.Angle_deg angle(start=angle_0) "Bus voltage angle";
+  parameter SI.PerUnit v_0=1 "Voltage magnitude (pu)"
     annotation (Dialog(group="Power flow data"));
-  parameter Real angle_0=0 "Voltage angle (deg)"
+  parameter SI.Conversions.NonSIunits.Angle_deg angle_0=0 "Voltage angle"
     annotation (Dialog(group="Power flow data"));
-  parameter Real V_b=130 "Base voltage (kV)"
+  parameter SI.Voltage V_b=130e3 "Base voltage"
```
Comment (marcelofcastro) [inline OpenIPSL/Electrical/Buses/BusExt.mo]: @dietmarw here you did not used the command to display the unit in kV. Was it intentional?

Comment (dietmarw) [reply OpenIPSL/Electrical/Buses/BusExt.mo]: Good catch. This was an oversight. Will fix it right away.

Comment (dietmarw) [reply OpenIPSL/Electrical/Buses/BusExt.mo]: Anything else of can this be merged?

File: OpenIPSL/Electrical/Banks/PSSE/SVC/SVC.mo [hunk 2]
```diff
@@ -24,39 +24,39 @@ model SVC "On bus 10106 & 10114"
...
   parameter Real T4 "Time constant (s)";
   parameter Real T5=0.03 "Time constant of thyristor bridge (s)";
-  parameter Real Vmax;
-  parameter Real Vmin;
-  parameter Real Vov=0.5 "Override voltage (pu)";
-  parameter Real Sbase "Base power of the bus (MVA)";
+  parameter SI.PerUnit Vmax;
+  parameter SI.PerUnit Vmin;
+  parameter SI.PerUnit Vov=0.5 "Override voltage (pu)";
+  parameter SI.ReactivePower Sbase(displayUnit="MVA") "Base power of the bus";
```
[the remark this answers is not in the document]
Comment (dietmarw) [reply OpenIPSL/Electrical/Banks/PSSE/SVC/SVC.mo]: yes :-)

**Pull request merged**